packages/plugin-core/src/test/suite-integ/MoveNoteCommand.test.ts

```diff
@@ -516,41 +530,174 @@ suite("MoveNoteCommand", function () {
+        vaultNoteAsserter(vault1).noMatchByFName("foo");
+
+        // In new vault
+        vaultNoteAsserter(vault2).matchByFName("foo");
+        await vaultAsserter(vault2).match(["foo.md"]);
+
+        done();
+      },
+    });
+  });
+
+  test("bulk-move: move 2 notes from different vaults to new vault", (done) => {
+    runLegacyMultiWorkspaceTest({
+      ctx,
+      preSetupHook: async ({ wsRoot, vaults }) => {
+        await ENGINE_HOOKS_MULTI.setupBasicMulti({ wsRoot, vaults });
+      },
+      onInit: async ({ engine, vaults, wsRoot }) => {
+        function vaultAsserter(vault: DVault) {
+          return new VaultAsserter({ wsRoot, vault });
+        }
+        function vaultNoteAsserter(vault: DVault) {
+          return new NoteAsserter({
+            workspaceVault: { wsRoot, vault },
+            notes: engine.notes,
+          });
+        }
+
+        const notes = engine.notes;
+        const vault1 = vaults[0];
+        const vault2 = vaults[1];
+        const vault3 = vaults[2];
+
+        const fooNote = NoteUtils.getNoteByFnameV5({
+          fname: "foo",
+          notes,
+          vault: vault1,
+          wsRoot,
+        }) as NoteProps;
+
+        await VSCodeUtils.openNote(fooNote);
+        const cmd = new MoveNoteCommand();
+
+        // @ts-ignore
+        VSCodeUtils.showQuickPick = async () => {
+          return "proceed";
+        };
+        await cmd.execute({
+          moves: [
+            {
+              oldLoc: {
+                fname: "foo",
+                vaultName: VaultUtils.getName(vault1),
+              },
+              newLoc: {
+                fname: "foo",
+                vaultName: VaultUtils.getName(vault3),
+              },
+            },
+            {
+              oldLoc: {
+                fname: "bar",
+                vaultName: VaultUtils.getName(vault2),
+              },
+              newLoc: {
+                fname: "bar",
+                vaultName: VaultUtils.getName(vault3),
+              },
+            },
+          ],
+        });
         expect(
-          await EngineTestUtilsV4.checkVault({
-            wsRoot,
-            vault: vault1,
-            nomatch: ["foo.md"],
-          })
-        ).toBeTruthy();
-        expect(
-          await EngineTestUtilsV4.checkVault({
-            wsRoot,
-            vault: vault2,
-            match: ["foo.md"],
-          })
-        ).toBeTruthy();
-        expect(
-          _.isUndefined(
-            NoteUtils.getNoteByFnameV5({
-              fname: "foo",
-              notes,
-              vault: vault1,
-              wsRoot,
-            })
+          VSCodeUtils.getActiveTextEditor()?.document.fileName.endsWith(
+            path.join("vault3", "foo.md")
           )
         ).toBeTruthy();
+
+        // Check that the files are not in old vaults anymore
+        await vaultAsserter(vault1).noMatch(["foo.md"]);
+        await vaultAsserter(vault2).noMatch(["bar"]);
```

Comment:
Left a separate comment regarding the fluent interface, so strictly commenting about readability here.

I think this could be more readable (and be consistent with what we are currently doing) if it looks something like:

```
const fooExists = await VaultTestUtils.noteExists({ vault: vault1, notes: ["foo"] });
expect(fooExists).toBeTruthy();
```

Reply:
```
const fooExists = await VaultTestUtils.noteExists({ vault: vault1, notes: ["foo"] });
expect(fooExists).toBeTruthy();
```

Unfortunately, this won't get us a descriptive failure message. Per other comments, just refactored to assertions to look as other tests in this file.